x/pairing/keeper/staking.go

```diff
@@ -166,9 +143,14 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, creator, chainID string, amount s
 	}
 
 	k.epochStorageKeeper.AppendStakeEntryCurrent(ctx, chainID, stakeEntry)
-	appended := false
 
-	details = append(details, utils.Attribute{Key: "effectiveImmediately", Value: appended})
+	err = k.dualStakingKeeper.Delegate(ctx, senderAddr.String(), senderAddr.String(), chainID, amount)
+	if err != nil {
+		utils.LavaFormatWarning("provider self delegation failed", err,
+			details...,
+		)
+	}
+
```

Comment:
let's emit the event only if err == nil, i know it wasnt like this before but this is a mistake (the event function also prints so it can be misleading)

Reply:
done